src/aws_credentials_env.erl

```diff
@@ -41,6 +41,12 @@ fetch(_Options) ->
                                   is_binary(T) ->
             {ok, aws_credentials:make_map(?MODULE, K, S, T), infinity};
 
+        {K, S, undefined, R} when is_binary(K),
+                                  is_binary(S),
+                                  is_binary(R) ->
+            Credentials = aws_credentials:make_map(?MODULE, K, S),
+            {ok, maps:put(region, R, Credentials), infinity};
```

Comment:
I do believe this breaks the abstraction  a bit of `aws_credentials:make_map/x` and probably would have preferred arity where if `Token` is `undefined`, it's simply not set but I won't block the PR because of it :+1: